tools/nni_cmd/launcher_utils.py

```diff
@@ -266,35 +266,14 @@ def validate_pai_config_path(experiment_config):
     '''validate paiConfigPath field'''
     if experiment_config.get('trainingServicePlatform') == 'pai':
         if experiment_config.get('trial', {}).get('paiConfigPath'):
-            # validate the file format of paiConfigPath, ensure it is yaml format
+            # check if commands exist
             pai_config = get_yml_content(experiment_config['trial']['paiConfigPath'])
-            if experiment_config['trial'].get('image') is None:
-                if pai_config.get('prerequisites', [{}])[0].get('uri') is None:
-                    print_error('Please set image field, or set image uri in your own paiConfig!')
-                    exit(1)
-                experiment_config['trial']['image'] = pai_config['prerequisites'][0]['uri']
-            if experiment_config['trial'].get('gpuNum') is None:
-                if pai_config.get('taskRoles', {}).get('taskrole', {}).get('resourcePerInstance', {}).get('gpu') is None:
-                    print_error('Please set gpuNum field, or set resourcePerInstance gpu in your own paiConfig!')
-                    exit(1)
-                experiment_config['trial']['gpuNum'] = pai_config['taskRoles']['taskrole']['resourcePerInstance']['gpu']
-            if experiment_config['trial'].get('cpuNum') is None:
-                if pai_config.get('taskRoles', {}).get('taskrole', {}).get('resourcePerInstance', {}).get('cpu') is None:
-                    print_error('Please set cpuNum field, or set resourcePerInstance cpu in your own paiConfig!')
-                    exit(1)
-                experiment_config['trial']['cpuNum'] = pai_config['taskRoles']['taskrole']['resourcePerInstance']['cpu']
-            if experiment_config['trial'].get('memoryMB') is None:
-                if pai_config.get('taskRoles', {}).get('taskrole', {}).get('resourcePerInstance', {}).get('memoryMB', {}) is None:
-                    print_error('Please set memoryMB field, or set resourcePerInstance memoryMB in your own paiConfig!')
-                    exit(1)
-                experiment_config['trial']['memoryMB'] = pai_config['taskRoles']['taskrole']['resourcePerInstance']['memoryMB']
-            if experiment_config['trial'].get('paiStoragePlugin') is None:
-                if pai_config.get('extras', {}).get('com.microsoft.pai.runtimeplugin', [{}])[0].get('plugin') is None:
-                    print_error('Please set paiStoragePlugin field, or set plugin in your own paiConfig!')
-                    exit(1)
-                experiment_config['trial']['paiStoragePlugin'] = pai_config['extras']['com.microsoft.pai.runtimeplugin'][0]['plugin']
+            if pai_config.get('taskRoles', {}).get('taskrole', {}).get('commands', [])[0] is None:
+                print_error('Please set paiStoragePlugin field, or set plugin in your own paiConfig!')
+                exit(1)
+            experiment_config['trial']['command'] = pai_config['taskRoles']['taskrole']['commands'][0]
```

Comment:
Seems that the support of PAI configuration is quite limited:

* Taskrole with name `taskrole` needs to exist.
* Only commands in `taskrole` will be wrapped with trial keeper.
* Multi-line commands are not supported.

Better states that in docs.

Reply:
Yes, uses need to provide `commands` field in config file, which is under `taskrole` field.